lib/Doctrine/DBAL/Types/DecimalType.php

```diff
@@ -49,6 +49,6 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
-        return (null === $value) ? null : $value;
+        return (null === $value) ? null : (string) $value;
```

Comment:
I had to modify this type because [this test](https://github.com/doctrine/dbal/blob/master/tests/Doctrine/Tests/DBAL/Functional/TypeConversionTest.php#L66) was still failing. This is because `SQLite3` returns numeric values as `float` for numbers that can be represented as a floating-point value without loosing precision, and as `string` for others.

I'm not sure if this line should be updated or if the test should be modified instead, but I suppose this should be all right as the Decimal type is always supposed to return a string.